src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -257,6 +257,29 @@
           request-cert)
         (log/warn (i18n/trs "Request is missing a certificate for an endpoint that requires a certificate."))))))
 
+(schema/def Certnames [schema/Str])
+
+(schema/defn handle-bulk-cert-signing
+  [request
+   _ca-settings :- ca/CaSettings]
+  (let [json-body (try-to-parse (:body request)) 
+        certnames (:certnames json-body)] 
+    (try+ 
+     (schema/validate certnames json-body)
+     (catch [:type :schema.core/error] {:keys [error]}
+       (throw+ {:kind :bad-request
+                :msg (format "Invalid input: %s" error)})))
```

Comment:
I think you can avoid the `try/catch` here and just allow the exception to propagate upward.